src/gui/org/deidentifier/arx/gui/view/impl/menu/DialogOrderSelection.java

```diff
@@ -199,7 +199,7 @@ private boolean isValidDataType(DataType<?> type, String[] values) {
      * @param charset TODO
      * @return
      */
-    private String[] loadArray(String file, Charset charset) {
+    private ArrayList<String> loadFile(String file, Charset charset) {

```

Comment:
Included the value existance check during file loading. If a value is not present within the attribute's domain the file loading is aborted. As a result the _retainAll_ call is not needed anymore.